src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,6 +514,128 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isCurrentProcessNet8Build, bool isVerbose)
+        {
+            string targetFramework = await GetTargetFrameworkAsync();
+            if (string.Equals(SetHostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isCurrentProcessNet8Build)
+                {
+                    PrintVerboseForHostSelection(isVerbose, "out-of-process");
+                    return false;
+                }
+            }
+            else if (string.Equals(SetHostRuntime, InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isCurrentProcessNet8Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+                {
+                    // Need this check in place so that any child process that is run for inproc8 host does not cause recursive calling and spawn another child process
+                    if (targetFramework == "net8.0")
+                    {
+                        PrintVerboseForHostSelection(isVerbose, InProc8HostRuntime);
+                        return true;
+                    }
+                }
+                else
+                {
+                    throw new CliException($"Invalid config for running {InProc8HostRuntime} host.");
+                }
+            }
+            else if (string.Equals(SetHostRuntime, InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (!isCurrentProcessNet8Build)
+                {
+                    return false;
+                }
+
+                PrintVerboseForHostSelection(isVerbose, InProc6HostRuntime);
+                return true;
+            }
+            else
+            {
+                throw new CliException($"Invalid host runtime '{SetHostRuntime}'. Valid values are 'default', '{InProc8HostRuntime}', '{InProc6HostRuntime}'.");
+            }
+            return false;
+        }
+
+        private void PrintVerboseForHostSelection(bool isVerbose, string hostRuntime)
+        {
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Selected {hostRuntime} host"));
+            }
+        }
+
+        private async Task<string> GetTargetFrameworkAsync()
+        {
+            var functionAppRoot = ScriptHostHelpers.GetFunctionAppRootDirectory(Environment.CurrentDirectory);
+
+            string targetFramework = "";
+
+            string projectFilePath = ProjectHelpers.FindProjectFile(functionAppRoot);
+            if (projectFilePath != null)
+            {
+                targetFramework = await DotnetHelpers.DetermineTargetFramework(Path.GetDirectoryName(projectFilePath));
+            }
+            return targetFramework;
+        }
+
+        private bool ShouldLaunchOutOfProcFromWorkerRuntime()
```

Comment:
We shouldn't have any scenarios where out of proc would be launched from this instance.